code/game/gamemodes/shadowling/shadowling.dm

```diff
@@ -201,11 +201,23 @@ Made by Xhuis
 /datum/game_mode/shadowling/check_finished()
 	var/shadows_alive = 0 //and then shadowling was kill
 	for(var/datum/mind/shadow in shadows) //but what if shadowling was not kill?
-		if(!istype(shadow.current,/mob/living/carbon/human) && !istype(shadow.current,/mob/living/simple_animal/ascendant_shadowling))
+		if(!ishuman(shadow.current) && !istype(shadow.current,/mob/living/simple_animal/ascendant_shadowling))
 			continue
 		if(shadow.current.stat == DEAD)
 			continue
 		shadows_alive++
+		if(shadow.special_role == SPECIAL_ROLE_SHADOWLING && config.shadowling_max_age)
+			if(ishuman(shadow.current))
+				var/mob/living/carbon/human/H = shadow.current
+				if(!istype(H.species, /datum/species/shadow))
```

Comment:
This should be `if(H.get_species() != "Shadow")` (I realize magic strings are usually worse than an istype check, but this is the consistent way of doing it through the rest of the code, and we'll probably need to overhaul get_species at some point)